azurelinuxagent/ga/exthandlers.py

```diff
@@ -374,11 +373,16 @@ def _extension_processing_allowed(self):
         return True
 
     def handle_ext_handlers(self, etag=None):
-        if self.ext_handlers.extHandlers is None or \
-                len(self.ext_handlers.extHandlers) == 0: # pylint: disable=len-as-condition
+        if not self.ext_handlers.extHandlers:
             logger.verbose("No extension handler config found")
             return
 
+        if self.last_etag == etag:
```

Comment:
Moving the check one level up to ensure we dont check for dependsOn status unless the incarnation changes

Reply:
Move it one more level up and do it along with ` if self._extension_processing_allowed():`?

Reply:
Done